File: php/leetcode/strings.php
<?php
declare(strict_types=1);
// TOPIC: String LeetCode Problems | php strings.php

// TODO 1: Longest Substring Without Repeating Characters — LC #3
//   function lengthOfLongestSubstring(string $s): int
//   // Sliding window + hash map of char → last index

// TODO 2: Valid Anagram — LC #242
//   function isAnagram(string $s, string $t): bool
//   // count_chars($s, 1) and count_chars($t, 1) must match

// TODO 3: Group Anagrams — LC #49
//   function groupAnagrams(array $strs): array
//   // Key = sorted chars (str_split + sort + implode)

// TODO 4: Longest Palindromic Substring — LC #5
//   function longestPalindrome(string $s): string
//   // Expand around center; track longest

// TODO 5: Minimum Window Substring — LC #76
//   function minWindow(string $s, string $t): string
//   // Sliding window; $need and $have char counts

// TODO 6: Encode and Decode Strings — LC #271
//   function encode(array $strs): string   // "4#word3#foo"
//   function decode(string $s): array      // parse length prefix

// TODO 7: Find All Anagrams in a String — LC #438
//   function findAnagrams(string $s, string $p): array
//   // Fixed sliding window; compare freq arrays

File: swift/leetcode/strings.swift
// TOPIC: String Problems (LeetCode) | swift strings.swift
// Run: swift strings.swift

// TODO 1: LC #3 — Longest Substring Without Repeating Characters
// Sliding window with a Set to track characters in the current window.
// func lengthOfLongestSubstring(_ s: String) -> Int

// TODO 2: LC #242 — Valid Anagram
// Sort both strings and compare, or use a frequency dictionary.
// func isAnagram(_ s: String, _ t: String) -> Bool

// TODO 3: LC #49 — Group Anagrams
// Group strings that are anagrams of each other; use sorted string as key.
// func groupAnagrams(_ strs: [String]) -> [[String]]

// TODO 4: LC #5 — Longest Palindromic Substring
// Expand around center for each character and each pair; track the longest found.
// func longestPalindrome(_ s: String) -> String

// TODO 5: LC #76 — Minimum Window Substring
// Sliding window with two frequency maps; shrink left when all chars covered.
// func minWindow(_ s: String, _ t: String) -> String

// TODO 6: LC #271 — Encode and Decode Strings
// Encode array of strings to single string; decode back (use length-prefix scheme).
// func encode(_ strs: [String]) -> String
// func decode(_ s: String) -> [String]

// TODO 7: LC #438 — Find All Anagrams in a String
// Sliding window of length p; compare frequency maps at each position.
// func findAnagrams(_ s: String, _ p: String) -> [Int]
